BUILD.md

```diff
@@ -0,0 +1,170 @@
+be installed by the preferred native way (e.g. via Package manager or compiling
+from source).
+
+Step by step procedure:
+
+ * On Windows or Mac OS X:
+   * Download the [FreeOrionSDK v8] from the FreeOrionSDK respository releases.
+ * Linux and other Operating Systems:
+   * Install build and runtime dependencies by the preferdded native way.
+ * Create a project directory `freeorion-project`.
+ * On Windows:
+   * Unzip the SDK archive contents into the project directory.
+   * Execute the `bootstrap.bat` within the project directory. This will clone
+     the FreeOrion repository and place the dependencies at the correct place.
+ * On Mac OS X:
+   * Mount the SDK DMG image.
+   * Copy the contents of the mounted image into the project directory.
+   * Execute the `bootstrap.command` within the project directory. This will
+     clone the FreeOrion repository and place the dependencies at the correct
+     place.
+ * On Linux and other Operating Systems:
+   * Navigate into the project directory.
+   * Clone the project via Git:
+     ```
+     git clone https://github.com/freeorion/freeorion.git
+     ```
+
+This will leave you with the latest development branch `master` and the
+FreeOrion source code in:
+
+ * `freeorion-project/FreeOrion/` on Windows.
+ * `freeorion-project/` on Mac OS X.
+ * `freeorion-project/freeorion/` on Linux and other Operating Systems.
+
+This directory will be referred to _source directory_ in the rest of the
+document.
+
+To build a specific release check out the desired version with `git checkout`:
+
+```
+git checkout vxxx
+```
+
+Where `vxxx` indicates the desired version.  A list of all available version
+can be listet by invoking `git tagi -l`, where release indicated by a leading
+`v` followed by the release version number.
+
+
+# Compiling FreeOrion
+
+FreeOrion uses CMake as build system on most platforms. On Windows a manually
+maintained Visual Studio Project is used for building.
+
+Step by step procedure:
+
+ * Enter the _source directory_.
+ * On Windows:
+   * Open `msvc2015\FreeOrion.sln` with Visual Studio.
+   * Compile the whole project by selecting the `Build` -> `Build Solution`
+     menu entry.
+
+ * On Mac OS X:
+   * Create a `build` directory, which will contain all compile FreeOrion
+     build artifacs.
+   * Change into the `build` directory on the command line.
+   * Execute cmake to generate a Xcode project file:
+
+     ```
+     cmake -GXcode ..
+     ```
+   * Open `FreeOrion.xcodeproj` with Visual Studio
+   * Compile the whole project by selecting the `ALL_BUILD` scheme and
+     pressing 'Command' + 'B'.
+
+ * On Linux and other Operating Systems
+   * Create a `build` directory, which will contain all compile FreeOrion
+     build artifacs.
+   * Change into the `build` directory on the command line.
+   * Execute cmake to generate Makefiles:
+
+     ```
+     cmake ..
+     ```
+   * Compile the whole project by calling `make` within the build directory.
```

Comment:
I get the impression from forum posts that for many people FO has been their first motivator to learn to compile, and it might be helpful to mention the ```-j n``` option for make.

Reply:
Added a note for the `--jobs` parameter.  Please check out if this fits with what you expect.